include/seastar/coroutine/exception.hh

```diff
@@ -68,7 +68,12 @@ namespace coroutine {
 /// ```
 struct exception {
     std::exception_ptr eptr;
-    explicit exception(std::exception_ptr eptr) : eptr(std::move(eptr)) {}
+
+    explicit exception(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
+    explicit exception(const std::exception_ptr& eptr) noexcept : eptr(eptr) {}
+
+    explicit exception(std::derived_from<std::exception> auto&& e) noexcept : eptr(std::make_exception_ptr(std::move(e))) {
```

Comment:
@nyh it can in theory (well, it could before this change). The question is if and why we want to support it.

Reply:
Because C++ is not Java, so exceptions do not **have** to derive from std::exception. It is true that std::exception has benefits (such as `what()` method that our operator<<(std::exception_ptr) can use), but since C++ doesn't absolutely require std::exception, I think neither should we. Especially since in the past we didn't, so why start now?

Reply:
@avikivity what's your take on this?
I can bring back support for make_exception_ptr on anything that's not an exception_ptr including, but not limiting to std::exception's.  This would still fix the original issue of masking the original exception_ptr and maintain compatibility with `throw` and `make_exception_future`.  I don't like carrying non-exception-type values in exceptional futures, but the language supports it... :-\

Reply:
@avikivity that's a different issue. Even if we didn't treat exception_ptr differently from other types, my question is why require that those other types must inherit from std::exception - since this is not a C++ requirement.